### PR TITLE
Feature/display hourly forecast

### PR DESCRIPTION
**Pull request by JanGiese**
Implemented display of hourly forecast.

- Added config variable forecast_hourly, which defaults to false
- Changed function names slightly to reflect usage for daily and hourly
- Changed distance for display of hours and minutes

### CODE REVIEW

**Comment by pkissling** [comment]
hi @JanGiese thanks for the PR. i just started an extended vacation where i didn't even bring a laptop. i'm afraid this PR has to wait for review until my vacation is over in january

File: src/styles.ts
Comment (pkissling) [inline src/styles.ts:64]: to not waste space if a user uses the regular `daily` instead of `hourly` forecast, i'd propose to make this change conditional to only change the column width from `2rem` to `2.5rem` if `forecast_hourly` is `true`

File: src/clock-weather-card.ts
Comment (pkissling) [inline src/clock-weather-card.ts:207]: this does not take into account a customisable `time_format`. if a user defines `time_format: 12` in the config, the time should be displayed as e.g. `12:00 AM`

File: src/clock-weather-card.ts
Comment (pkissling) [inline src/clock-weather-card.ts:188]: what happens if a user defines `forecast_hourly: true`, but the weather integration provides `daily` values? have you tried that?

**Review by pkissling — CHANGES_REQUESTED** [review]

File: src/clock-weather-card.ts
Comment (JanGiese) [reply src/clock-weather-card.ts:207]: makes sense. I have overlooked that setting. will add it

File: src/clock-weather-card.ts
Comment (pkissling) [inline src/clock-weather-card.ts:467]: there is an issue on how the aggregation works now. 
forecasts will be aggregated by hours, which means for example a forecast at 2pm for today will be combined with a forecast for 2pm tomorrow and later `calculateAverageForecast` will calculate the average of both.

this causes another problem where a maximum of 24 hours can be displayed, even if the weather provider delivers more than 24 hours

**Review by pkissling — CHANGES_REQUESTED** [review]

File: src/styles.ts
Comment (JanGiese) [reply src/styles.ts]: Was especcially needed when supporting the 12h display. Added it.

File: src/clock-weather-card.ts
Comment (JanGiese) [reply src/clock-weather-card.ts:489]: Thanks for the heads-up. I have added the date to the aggregation as well to fix this.

File: README.md
Comment (pkissling) [inline README.md:95]: was this deletion intentional?

File: README.md
Comment (pkissling) [inline README.md:116]: was this deletion intentional?

File: src/clock-weather-card.ts
Comment (pkissling) [inline src/clock-weather-card.ts:207]: boxing via `new Date(...)` not required, as `forecast.datetime` is already of type `Date`

Comment (pkissling) [reply src/clock-weather-card.ts:207]: additionally, to not have the formatting logic (`HH:mm` vs `h:mm aa`) in two different locations redundantly, you could extend the existing method `this.time()` for formatting:
```
  private time(date: Date = this.currentDate): string {
    return format(date, this.getTimeFormat() === '24' ? 'HH:mm' : 'h:mm aa');
  }
```

then you would simplify the whole line to: 
`const displayText = !hourly ? this.localize(`day.${forecast.datetime.getDay()}`) : this.time(forecast.datetime);`

File: src/types.ts
Comment (pkissling) [inline src/types.ts:41]: wouldn't `hourly_forecast` make more sense?

**Review by pkissling — CHANGES_REQUESTED** [review]

**Comment by pkissling** [comment]
Fixes #64

File: README.md
Comment (JanGiese) [reply README.md:109]: No, I have overlooked it while merging. Have added it again.

File: README.md
Comment (JanGiese) [reply README.md:130]: No, I have overlooked it while merging. Have added it again.

File: src/clock-weather-card.ts
Comment (JanGiese) [reply src/clock-weather-card.ts]: Addded it, thanks for the heads up

File: src/types.ts
Comment (JanGiese) [reply src/types.ts]: changed it accordingly

File: README.md
Comment (pkissling) [inline README.md:131]: `hide_date` should not be deleted

File: README.md
Comment (pkissling) [inline README.md:129]: this entry should not be duplicated

**Review by pkissling — CHANGES_REQUESTED** [review]
looks good to me (besides the unintended changes in README.md). thanks for incorporating my suggestions. i'm happy to merge the PR once you fixed the documentation remarks rearding duplication of `hide_forecast_section` and deletion of `hide_date` are fixed from your side

**Comment by JanGiese** [comment]
Hi Patrick,
when I checked in the morning, it looked like in your screenshot, but now it looks like this:

image

**Comment by JanGiese** [comment]
Hi,
I have re-checked the logic and the range and indicator didn´t made sense, as they dependant on isToday. i have now changed the variable and parameter name to isNow and set that dependent on the hourly_forecast config entry, so the indicator is only shown on the "current" forecast.

**Review by pkissling — APPROVED** [review]

**Comment by pkissling** [comment]
looks good now. thanks for your contribution.
image
i fixed a small bug #169 where sometimes the temperature dot was missing (for example also in your screenshot, at time 00:00). but it was not related to your implementation
image

**Comment by pkissling** [comment]
i will create a release tomorrow

**Pull request merged**
